tests/handlers/botas/test_botas_shipped_advisory.py

```diff
@@ -98,11 +99,121 @@ def test_handle_isnt_allowed_by_internal_policy(self):
         self.assertTrue(db_event.state_reason.startswith(
             "This image rebuild is not allowed by internal policy."))
 
-    @patch.object(conf, 'dry_run', new=True)
-    @patch.object(conf, 'handler_build_allowlist', new={
-        'HandleBotasAdvisory': {
-            'image': {
-                'advisory_name': 'RHBA-2020'
+    def test_handle(self):
+        event = BotasErrataShippedEvent("test_msg_id", self.botas_advisory)
+        self.handler.allow_build = MagicMock(return_value=True)
+        self.handler._create_original_to_rebuilt_nvrs_map = \
+            MagicMock(return_value={"original_1": "some_name-1-12345",
+                                    "original_2": "some_name_2-2-2"})
+        nvr_to_digest = {
+            "original_1": "original_1_digest",
+            "some_name-1-12345": "some_name-1-12345_digest",
+            "original_2": "original_2_digest",
+            "some_name_2-2-2": "some_name_2-2-2_digest",
+        }
+        bundles = [{"bundle_path_digest": "original_1_digest"},
+                   {"bundle_path_digest": "some_name-1-12345_digest"},
+                   {"bundle_path_digest": "original_2_digest"},
+                   {"bundle_path_digest": "some_name_2-2-2_digest"}]
+        related_bundles = {"original_1_digest": [{"bundle_path_digest": "related_1_digest"}],
+                           "original_2_digest": [{"bundle_path_digest": "related_2_digest"}]}
+
+        image_by_digest = {
+            "related_1_digest": {"brew": {"build": "bundle1_nvr-1-1"}},
+            "related_2_digest": {"brew": {"build": "bundle2_nvr-1-1"}},
+        }
+        self.pyxis().get_manifest_list_digest_by_nvr.side_effect = lambda x: nvr_to_digest[x]
+        self.pyxis().get_operator_indices.return_value = []
+        self.pyxis().get_latest_bundles.return_value = bundles
+        # return bundles for original images
+        self.pyxis().get_bundles_by_related_image_digest.side_effect = lambda x, y: related_bundles[x]
+        self.pyxis().get_images_by_digest.side_effect = lambda x: [image_by_digest[x]]
+        self.handler.image_has_auto_rebuild_tag = MagicMock(return_value=True)
+        get_build = self.patcher.patch("freshmaker.kojiservice.KojiService.get_build")
```

Comment:
Could you use the context manager here so that this patch is automatically cleaned up after your test (pass or fail)?

Reply:
But it's done in `tearDown()`. It calls `self.patcher.unpatch_all()` after every test, in that method all patched objects are unpatched.